HW#2 – Pavel Belyaev

20eleven opened the pull request:
### In this project task I:
 - wrote 2 versions of the code - on **Flex** and on **Grid**
 - added **responsive** design, now it can be opened on different devices
 - implemented **cross-browser** compatibility
 - replaced the usual CSS with the **SASS** preprocessor (compiled it into the dist folder with the necessary prefixes)
 - used the **BEM** methodology
 - added edits from the HW#1
<br><br>

_I didn't separate the flex/grid layout from the main styles files, as it seems that it would be less convenient for use experience  in that case_

_For compilation I used a plugin in VSCode - "Live Sass Compiler"_

20eleven committed:
Add all GRID version files + gitignore

20eleven committed:
Add all FLEX verion files

20eleven committed:
Put the widget in the center of the screen

20eleven committed:
Change titles font size

20eleven committed:
Change auto for 1fr

20eleven committed:
Add shadow on widget

20eleven committed:
Add a new block for center alignment

20eleven committed:
Add background image and change center-block height value

20eleven committed:
Change max-width

20eleven committed:
Disable selection of text for titles

20eleven committed:
Fix the issue with small icons in ie11 on FLEX version

20eleven committed:
Fix the issue with display in ie11 on GRID version

20eleven committed:
Change background to original from mockup

blackbird01123 commented:
On screens with width less than 620px the cubes are no longer the cubes (or, more precisely, squares). Either keep the proportions, or keep the same height for the rectangles on hover - now when hovering over the cubes on small screens there is jumping behavioк (both flexbox and grid).

blackbird01123 commented on flexVersion/index.html:
```diff
@@ -0,0 +1,255 @@
+<body>
+
+   <article class="weather-widget">
+      <div class="weather-widget__center-block"></div>
```
Try to avoid using elements for styling purposes. Consider using `weather-widget` as a container and apply necessary flexbox styling to center its content (`weather-widget__body`).

20eleven replied:
I used the weather-widget as a container to center its contents, but did not use flexbox, because in this task (mean the grid version) I wanted to practice without using flex (only grid or standard properties)

blackbird01123 replied:
I mean, don't use empty elements for styling (`weather-widget__center-block`). And of course you could do the grid part of homework without using flexbox, sorry if I confused you. Just try to use only necessary elements in template and do the rest with styles.

blackbird01123 commented on gridVersion/styles/style.scss:
```diff
@@ -0,0 +1,96 @@
+   .weather-widget__box {
+      display: grid;
+      grid-auto-flow: row;
+      grid-template: 20% 50% 30% / 1fr;
```
Why are you using 3 rows in `weather-widget__box`? If it's for styling purposes, consider using 2 rows and align the icon with margins or flexbox.

20eleven replied:
I fixed it, but I heard that using additional grid cells for alignment is considered quite good practice for flexibility

blackbird01123 commented on gridVersion/index.html:
```diff
@@ -0,0 +1,255 @@
+   <title>Cubes</title>
+   <link rel="stylesheet" href="./dist/css/style.min.css">
+   <link href="https://fonts.googleapis.com/css2?family=Lato:wght@700&display=swap" rel="stylesheet">
+   <style type="text/css">
+      body {
+         background-image: url(./img/bgPicture/bluredBG.png);
+         background-repeat: no-repeat;
+         background-position: center center;
+         background-size: cover;
+      }
+   </style>
```
It might be better to move internal styles to an external stylesheet.

blackbird01123 commented:
On small screens in IE11 there is the issue with the proportions of icons and text in `weather-widget__week-temperature` (grid):
![image](https://user-images.githubusercontent.com/71438780/95369105-a7bf6680-08df-11eb-80e2-de1ec9a2a601.png)

blackbird01123 commented:
In general, very well 👍 Keep it up!

20eleven committed:
Fix jumping behavior on small screens and fix the display of icons on FLEX version

20eleven committed:
Fix jumping behavior on small screen in GRID version

20eleven committed:
Fix the display of icons on GRID

20eleven committed:
Move internal styles to an external stylesheet from html

20eleven committed:
Change the centering method of the widgets

20eleven committed:
Change 3 rows for 2 rows and margin in weather-widget__box on GRID

20eleven committed:
Change auto for 1fr on GRID(for better display on ie11)

20eleven committed:
Fix display in ie11 on GRID

20eleven commented:
To keep the aspect ratio using only css, need to use too many media queries, so I fixed the display on widened cells and now they do not jump and look more decent. Also corrected the display of icons

**UPD: already fixed that**
_But there is a bug (in GRID versoin and only in Chrome) - if on screen less than 620px scroll down a little bit, about 5-10px, but so that the large icon is completely visible, then when using hover, the first cell has a jumping behavior, only 1st 
upper item not the rest 🤔 but when the box is fully open or the top border of the screen over the icon, then everything is fine_

20eleven committed:
Fix a bug in GRID version with jumping behavior of the first cell

blackbird01123 reviewed (commented):
The issue with icons in IE11 is still there (grid version). Images proportions are wrong (see how they look in Chrome to compare). Play with `height` property for `img`, maybe setting height to `auto` could help (don't forget to remove `max-height` in this case).
![image](https://user-images.githubusercontent.com/71438780/97175935-49322d80-17a5-11eb-8c4a-20d3ba5ab666.png)

20eleven committed:
Fix the display of icons on GRID version

20eleven commented:
Fixed the display of icons (hover) in ie11 on grid version. Now they look much prettier

**Thanks a lot for your feedbacks**, they will undoubtedly help me better understand the nuances of layout and work in general 😁

blackbird01123 commented:
Thank you! I'm so glad you find these comments useful. You're doing really great too 😀
